cmd/mdatagen/metadata-schema.yaml

```diff
@@ -1,6 +1,17 @@
-# Required: type of the receiver.
+# Required: The type of the component.
 type:
 
+# Required: A high-level view of the development status and use of this component
+status:
+  # Required: The class of the component (For example receiver)
+  class: <receiver|processor|exporter|cmd|connector|extension>
```

Comment:
The code doesn't check for any specific enum values - Just that it exists. Should we make the schema documentation less strict and allow any string?

Reply:
The code currently doesn't, because we're in the process of adopting this across the board. See #2123 for the effort and https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21275 and https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21274 for the first efforts.

Reply:
I'm not sure the first link is what you intended but the last two look great! I'll definitely include the specific values in the documentation then.

Reply:
Sorry I meant #21213